### DISCUSSION TITLE
Feature request: packages/react-instantsearch-core/src/components/InstantSearch.tsx should have a prop to specify the loading component

### DISCUSSION THREAD

**Discussion by sbdchd** [algolia/instantsearch#5981] [Ideas]
Right now it renders `null` while it's waiting to render something

Instead it would be possible to customize this loading state

The problem with rendering `null` is that it causes a flash as the component first renders, then the search starts, and then it renders again actually showing the children

**Comment by Haroenv** [comment]
What would you want to render? The actual InstantSearch children or some very short lived loading state? it's right that this could cause a flash without server side rendering

**Reply by sbdchd** [reply]
Essentially a copy of the the stuff that is connected to the instant search context

So in my case I'd just have the search input that doesn't call into instant search, so there wouldn't be a flash

Ideally we wouldn't need to have a skeleton / loading state, the null return wouldn't exist
